app/main.py

```diff
@@ -1,3 +1,14 @@
-def cache(func):
-    # Write your code here
-    pass
+def cache(func: callable) -> callable:
+    cached = {}
+
+    def inner(*args: int | float) -> int | list:
```

Comment:
arguments and return value can have any type. use `Any` or `ParamSpec` with `TypeVar` instead